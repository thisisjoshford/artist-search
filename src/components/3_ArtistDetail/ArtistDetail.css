.Artist{
  height: 100%;

  img {
    width: 10rem;
  }
}