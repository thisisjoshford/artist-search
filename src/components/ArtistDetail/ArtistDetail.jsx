import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import styles from './ArtistDetail.css';


const ArtistDetail = ({ artistName, artistDetail }) => {
  const artistReleases = artistDetail.map((release) => (
    <Link 
      key={release.id} 
      to={`/${artistName}/${release.title}/${release.id}/`} 
    >
      <div key={release.id} className={styles.Artist}> 
        <img src={release.cover_art} />
        <h3>{release.title}</h3>
      </div>
      
    </Link>
  ));

  return (
    <>
      <h3>Releases for: {artistName}</h3>
      {artistReleases}
    </>
    
  );
};

ArtistDetail.propTypes = {
  artistName: PropTypes.string.isRequired,
  artistDetail: PropTypes.array.isRequired
};

export default ArtistDetail;
