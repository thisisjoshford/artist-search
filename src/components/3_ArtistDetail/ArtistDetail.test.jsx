import React from 'react';
import { shallow } from 'enzyme';
import ArtistDetail from './ArtistDetail';

describe('Search component', () => {
  it('renders Search', () => {
    const artistDetail = [
      {
        id: '1234',
        title: 'album 1',
        cover_art: 'http://coverarturl.jpg'
      },
      {
        id: '12345',
        title: 'album 2',
        cover_art: 'http://coverarturl.jpg'
      }
    ];

    const wrapper = shallow(
      <ArtistDetail artistName='Tool' artistDetail={artistDetail}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
