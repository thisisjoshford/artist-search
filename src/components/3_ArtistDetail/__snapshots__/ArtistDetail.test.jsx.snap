// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search component renders Search 1`] = `
<Fragment>
  <h3>
    Releases for: 
    Tool
  </h3>
  <Link
    key="1234"
    to="/Tool/album 1/1234/"
  >
    <div
      className="Artist"
      key="1234"
    >
      <img
        src="http://coverarturl.jpg"
      />
      <h3>
        album 1
      </h3>
    </div>
  </Link>
  <Link
    key="12345"
    to="/Tool/album 2/12345/"
  >
    <div
      className="Artist"
      key="12345"
    >
      <img
        src="http://coverarturl.jpg"
      />
      <h3>
        album 2
      </h3>
    </div>
  </Link>
</Fragment>
`;
